pkg/ingress/controller/controller.go

```diff
@@ -558,7 +627,26 @@ func (c *Controller) ensureIngress(ing *nwv1beta1.Ingress) error {
 		log.WithFields(log.Fields{"sgID": sgID, "ingress": key}).Info("ensured security group")
 	}
 
-	listener, err := c.osClient.EnsureListener(name, lb.ID)
+	// Convert kubernetes secrets to barbican ones
+	var secretRefs []string
+	for _, tls := range ing.Spec.TLS {
+		if c.osClient.Barbican == nil {
```

Comment:
If I understand this correct, this should go in form of
```
if len(ing.Spec.TLS) > 0 && c.osClient.Barbican == nil {
}
```
To the very beginning of the function.

There is no need to `EnsureLoadBalancer` and Managing Security Groups if this code check will fail.

Reply:
Make sense, thanks!